### ISSUE TITLE
How about we 'host' JsonRulesEngine in DotNet?

### ISSUE THREAD

**Issue by timiil** [ozanerturk/CSharpJsonRulesEngine#1]
Hello,  just have this idea , but i am not sure yet it can be done:

use something like <a>https://github.com/microsoft/ClearScript</a> to host the <a>https://github.com/CacheControl/json-rules-engine</a> into a dotnet application,  so we can still use the 'json-rules-editor' and its json grammer, also we can use it in our dotnet app.

**Comment by timiil** [comment]
Another way may be <a>https://github.com/microsoft/RulesEngine</a>, we just create an application that translate the json from 'json-rules-engine' foramt into the 'RulesEngine' format, so we can still use the 'json-rules-editor' and its json grammer, also we can use it in our dotnet app.

**Comment by ozanerturk** [comment]
Yeah, it could have made like that. Firstly, this port uses same json schema so you can use same editor. Still have some missing part like prioritizing etc. 

Using script engine creates another dependency. I don't know how much weight it would add but I think it worths to try.

Microsoft's rules engine is not flexible like json rules engine and converting cost should be considered in this kind of application. 
__
I am thinking about adding distributed persistency layer to store rules so it can scale well. Redis's pub/sub would fit easily.
